android/jni/com/mapswithme/maps/MapManager.cpp

```diff
@@ -94,6 +95,13 @@ Java_com_mapswithme_maps_downloader_MapManager_nativeIsLegacyMode(JNIEnv * env,
   return !version::IsSingleMwm(GetStorage().GetCurrentDataVersion());
 }
 
+// static native boolean nativeNeedMigrate();
+JNIEXPORT jboolean JNICALL
+Java_com_mapswithme_maps_downloader_MapManager_nativeNeedMigrate(JNIEnv * env, jclass clazz)
+{
+  return platform::migrate::NeedMigrate();
```

Comment:
Разве нет? Фреймворк теперь в фоне не создается?

> On Mar 25, 2016, at 18:08, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/MapManager.cpp <https://github.com/mapsme/omim/pull/2512#discussion_r57439253>:
> 
> > @@ -94,6 +95,13 @@ Java_com_mapswithme_maps_downloader_MapManager_nativeIsLegacyMode(JNIEnv * env,
> >    return !version::IsSingleMwm(GetStorage().GetCurrentDataVersion());
> >  }
> >  
> > +// static native boolean nativeNeedMigrate();
> > +JNIEXPORT jboolean JNICALL
> > +Java_com_mapswithme_maps_downloader_MapManager_nativeNeedMigrate(JNIEnv * env, jclass clazz)
> > +{
> > +  return platform::migrate::NeedMigrate();
> Потому что isSingleMwm требует инициализированный storage. В местах вызова его ещё нет и не желательно поднимать.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2512/files/2ed1aa65936a4f781a177777c26bb277038f0b0b#r57439253>